catroid/src/org/catrobat/catroid/content/bricks/PlaySoundAndWaitBrick.java

```diff
@@ -0,0 +1,372 @@
+
+		playSoundAndWaitBrickSpinner.setAdapter(spinnerAdapterWrapper);
+
+		setSpinnerSelection(playSoundAndWaitBrickSpinner);
+
+		return view;
+	}
+
+	@Override
+	public View getViewWithAlpha(int alphaValue) {
+
+		if (view != null) {
+
+			View layout = view.findViewById(R.id.brick_play_sound_and_wait_layout);
+			Drawable background = layout.getBackground();
+			background.setAlpha(alphaValue);
+
+			TextView playSoundLabel = (TextView) view.findViewById(R.id.brick_play_sound_and_wait_label);
+			playSoundLabel.setTextColor(playSoundLabel.getTextColors().withAlpha(alphaValue));
+
+			Spinner lookbrickSpinner = (Spinner) view.findViewById(R.id.playsound_spinner);
+
+			ColorStateList color = playSoundLabel.getTextColors().withAlpha(alphaValue);
+			lookbrickSpinner.getBackground().setAlpha(alphaValue);
+			if (adapterView != null) {
+				((TextView) adapterView.getChildAt(0)).setTextColor(color);
+			}
+
+			this.alphaValue = alphaValue;
+		}
+
+		return view;
+	}
+
+	private ArrayAdapter<SoundInfo> createSoundAdapter(Context context) {
+		ArrayAdapter<SoundInfo> arrayAdapter = new ArrayAdapter<SoundInfo>(context,
+				android.R.layout.simple_spinner_item);
+		arrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+		SoundInfo dummySoundInfo = new SoundInfo();
+		dummySoundInfo.setTitle(context.getString(R.string.new_broadcast_message));
+		arrayAdapter.add(dummySoundInfo);
+		for (SoundInfo soundInfo : ProjectManager.getInstance().getCurrentSprite().getSoundList()) {
+			arrayAdapter.add(soundInfo);
+		}
+		return arrayAdapter;
+	}
+
+	@Override
+	public View getPrototypeView(Context context) {
+		View prototypeView = View.inflate(context, R.layout.brick_play_sound_and_wait, null);
+		Spinner playSoundSpinner = (Spinner) prototypeView.findViewById(R.id.playsound_spinner);
+		playSoundSpinner.setFocusableInTouchMode(false);
+		playSoundSpinner.setFocusable(false);
+		playSoundSpinner.setEnabled(false);
+
+		SpinnerAdapter playSoundSpinnerAdapter = createSoundAdapter(context);
+		playSoundSpinner.setAdapter(playSoundSpinnerAdapter);
+		setSpinnerSelection(playSoundSpinner);
+		return prototypeView;
+	}
+
+	@Override
+	public Brick clone() {
+		return new PlaySoundAndWaitBrick();
+	}
+
+	//for testing purposes:
+	public void setSoundInfo(SoundInfo soundInfo) {
+		this.sound = soundInfo;
+	}
+
+	@Override
+	public void onItemSelected(AdapterView<?> parent, View arg1, int position, long arg3) {
+		if (position == 0) {
+			sound = null;
+		} else {
+			sound = (SoundInfo) parent.getItemAtPosition(position);
+			oldSelectedSound = sound;
+		}
+		adapterView = parent;
+	}
+
+	@Override
+	public void onNothingSelected(AdapterView<?> arg0) {
+	}
+
+	@Override
+	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
+		sequence.addAction(sprite.getActionFactory().createPlaySoundAndWaitAction(sprite, sound));
```

Comment:
the standard ``PlaySoundAction`` should suffice. use 
``sprite.getActionFactory().createPlaySoundAction()`` instead